include/nbl/builtin/hlsl/radix_fft.hlsl

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_RADIX_FFT_INCLUDED_
+#define _NBL_BUILTIN_HLSL_RADIX_FFT_INCLUDED_
+
+#include <nbl/builtin/hlsl/complex.hlsl>
+
+
+namespace nbl
+{
+namespace hlsl
+{
+
+
+// Fast Fourier Transform
+template<typename T, uint32_t N>
+void fft_radix2(complex::complex_t<T> data[N], bool is_inverse)
```

Comment:
make the `is_inverse` a template parameter

`assert(N<gl_SubgroupSize)` 

btw the `data` need not be `data[N]`, each invocation comes with its own register value and you can cooperatively exchange via the use of `subgroupShuffleXor` (add it to https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/builtin/hlsl/glsl_compat/subgroup_shuffle.hlsl please).

I mean the original GLSL code is here
https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/builtin/glsl/subgroup/fft.glsl